### DISCUSSION TITLE
How to get previous pathname (the pathname that router.back() will go) in app router?

### DISCUSSION THREAD

**Discussion by tonys1997** [vercel/next.js#51555] [Help]
### Summary

I can get current pathname  by `usePathname`:
`import { usePathname, } from 'next/navigation'`

But can I get previous pathname?
This is the pathname that `router.back()` will go.

How can I get it?

thanks

**Comment by alirezasoltanian** [comment]
I have the same problem don't need router.back() I want just pathname of the previous page. are mean don't way for this problem?

**Reply by MuhammadYounus09** [reply]
To obtain the previous pathname, you can store the current pathname using the useEffect hook and update it whenever the route changes

**Reply by icyJoseph** [reply]
> This makes traversing the history a much less fragile proposition than with the older [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API). - https://developer.mozilla.org/en-US/docs/Web/API/Navigation

I had initially written a long form on why, for frameworks, to claim such a feature as "history traversal" is tricky! but I deleted it all, just to say that it is so, and we ought to deal with it. 

We can make a broad assumption though, that we are the only ones controlling navigation in our app, and with that, the usePathname or useParams (I prefer this one), that creates a stack of entries is the way to go.
